src/test/test-execute.c

```diff
@@ -76,6 +76,11 @@ static void test(Manager *m, const char *unit_name, int status_expected, int cod
         check(m, unit, status_expected, code_expected);
 }
 
+static int in_container(void) {
+        char *container = getenv("container");
+        return container != NULL && strcmp(container, "systemd-nspawn") == 0;
```

Comment:
instead of defining a new function for this, please just use "detect_container(NULL) > 0", it's defined in virt.h